shared/src/main/scala/amf/core/model/domain/DataModel.scala

```diff
@@ -91,11 +89,19 @@ class ObjectNode(override val fields: Fields, val annotations: Annotations) exte
 
   def addProperty(propertyOrUri: String, objectValue: DataNode, annotations: Annotations = Annotations()): this.type = {
     val property = ensurePlainProperty(propertyOrUri)
-    objectValue.set(DataNodeModel.Name, property)
+    overrideName(objectValue, property)
     addPropertyByField(createField(property), objectValue, annotations)
     this
   }
 
+  private def overrideName(objectValue:DataNode, newName:String) = {
+    objectValue.fields.getValueAsOption(DataNodeModel.Name).fold({
+      objectValue.synthesizeName(newName)
```

Comment:
although fold and match case are the same, is fold more readable than a match-case?

Reply:
In my opinion it is, but maybe is more common for others use the match. Anyway, I used fold instead of match for performance, to run pure functional witch is lazy.

Reply:
Hmm don't know it it's more performance as the fold method in an Option is implemented like:
```scala
@inline final def fold[B](ifEmpty: => B)(f: A => B): B =
    if (isEmpty) ifEmpty else f(this.get)
```
It is more of a "declarative" style I agree, won't make a fuss about this. Resolving this conversation.

Reply:
If I remember correctly the if statement is much more performant than the match structure.